### PR TITLE
LPS-144813 When using a workflow with parallel task, the join node is displayed as current node and past node

### PR DESCRIPTION
**Pull request by michaelmdelcavalcanti**
LPS: [LPS-144813](https://issues.liferay.com/browse/LPS-144813)

From what i could understand about the workflow business rule, the tasks are the only steps where the action of third parties is expected, consequently it is the only step in which the current nodes of the workflow will be, so I added a filter so that the current nodes only accept task values.
I also added a filter for the visited nodes, the join being the only one that has a different behavior, I added a filter to meet your specific conditions.

Example:
image

### CODE REVIEW

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
ci:reevaluate:1397401_576

**Comment by michaelmdelcavalcanti** [comment]
@naraandrade. about errors in relevant.
The FilterAllItemsListByAssigneeName test: I believe it is performing the reindex in the metrics and there it is opening the incorrect workflow, I performed the test before performing the reindex and observed the test opening the incorrect process, after performing the reindex he managed to open the correct one and then the test passed normally.
The CanIdentifyMultipleCurrentNode test: the track workflow is opening too fast and the track overflow is not loading all components resulting in the test failing.

File: modules/apps/portal-workflow/portal-workflow-instance-tracker-web/src/main/resources/META-INF/resources/js/components/WorkflowInstanceTracker.js [hunk 1]
```diff
@@ -135,6 +141,15 @@ export default function WorkflowInstanceTracker({workflowInstanceId}) {
 
 	const elements = nodes.concat(transitions);
 
+	useEffect(() => {
+		nodes.map((node) => {
+			if (node.data.current === true) {
```
Comment (alinedoleron) [inline modules/apps/portal-workflow/portal-workflow-instance-tracker-web/src/main/resources/META-INF/resources/js/components/WorkflowInstanceTracker.js:146]: Hi @michaelmdelcavalcanti, `node.data.current` value is always a boolean? If so, could you change this to the following, please?
```suggestion
			if (node.data.current) {
```

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Hello, @michaelmdelcavalcanti. Good job! I've tested the PR and it works. 
There's just one simple adjustment to do. Please, check the comment.
Thanks!

**Comment by alinedoleron** [comment]
Also, please, could you update the [ticket](https://issues.liferay.com/browse/LPS-144813) status to `In Review` and link this PR to it?

File: modules/apps/portal-workflow/portal-workflow-instance-tracker-web/src/main/resources/META-INF/resources/js/components/WorkflowInstanceTracker.js (hunk 1 above)
Comment (michaelmdelcavalcanti) [reply modules/apps/portal-workflow/portal-workflow-instance-tracker-web/src/main/resources/META-INF/resources/js/components/WorkflowInstanceTracker.js]: Hello Aline, fixed.

**Comment by alinedoleron** [comment]
ci:forward

**Review by alinedoleron — APPROVED** [review]

**Comment by alinedoleron** [comment]
ci:forward

**Comment by alinedoleron** [comment]
Hi @michaelmdelcavalcanti, please, could you run locally to check if these poshi test works on your machine? 

1. [WorkflowMetrics#ViewTopAsigneePerformance](https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/243272//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_WorkflowMetrics_ViewTopAsigneePerformance_)

2. [WorkflowMetricsAllItemsList#CanIdentifyMultipleCurrentNode](https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/281118//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_WorkflowMetricsAllItemsList_CanIdentifyMultipleCurrentNode_)

You need to run this command for each test. For example:
`ant -f build-test.xml run-selenium-test -Dtest.class=WorkflowMetrics#ViewTopAsigneePerformance` on your on you **liferay-portal** root. 

Please let me know if you need any help.
Thanks!

**Comment by michaelmdelcavalcanti** [comment]
Hello @alinedoleron, about the tests.
ViewTopAsigneePerformance: I didn't find a relationship with the solution, the test breaks when entering a script in metrics.
CanIdentifyMultipleCurrentNode: the track workflow is opening too fast and the track overflow is not loading all components resulting in the test failing.

**Comment by alinedoleron** [comment]
Thanks for the analysis, @michaelmdelcavalcanti!

Hi @anabuchmann, could you or another QA of workflow team have a look on this two failing tests, please? 
Thanks!

**Comment by anabuchmann** [comment]
Just started reviewing :)

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by michaelmdelcavalcanti** [comment]
@alinedoleron, new fix passed on tests, waiting for your review.

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by alinedoleron** [comment]
ci:forward

**Pull request closed without merging**